File: zchan_player_annotating/src/main/java/com/example/zchan_player_annotating/view/PlayerSurface.java
package com.example.zchan_player_annotating.view;

import android.content.Context;
import android.opengl.GLSurfaceView;
import android.util.AttributeSet;
import android.view.SurfaceHolder;

import javax.microedition.khronos.egl.EGLConfig;
import javax.microedition.khronos.opengles.GL10;

/**
 * Created by Mr.Chan
 * Time 2023-12-28
 * Blog https://www.cnblogs.com/Frank-dev-blog/
 */
public class PlayerSurface extends GLSurfaceView implements SurfaceHolder.Callback, GLSurfaceView.Renderer {

    private OnSurfaceListener onSurfaceListener;
    public void setOnSurfaceListener(OnSurfaceListener onSurfaceListener) {
        this.onSurfaceListener = onSurfaceListener;
    }
    public interface OnSurfaceListener {
        void onSurfaceCreated(SurfaceHolder holder);
    }
    public PlayerSurface(Context context, AttributeSet attrs) {
        super(context, attrs);
        setRenderer(this);
    }

    @Override
    public void surfaceCreated(SurfaceHolder holder) {
        if (onSurfaceListener != null) {
            onSurfaceListener.onSurfaceCreated(holder);
        }
    }

    @Override
    public void surfaceChanged(SurfaceHolder var1, int var2, int var3, int var4) {

    }

    @Override
    public void surfaceDestroyed(SurfaceHolder var1) {

    }



    @Override
    public void onSurfaceCreated(GL10 gl10, EGLConfig eglConfig) {

    }

    @Override
    public void onSurfaceChanged(GL10 gl10, int i, int i1) {
    }

    @Override
    public void onDrawFrame(GL10 gl10) {

    }
}

File: playerCore/src/main/java/com/example/playercore/PlayerSurface.kt
package com.example.playercore

import android.content.Context
import android.opengl.GLSurfaceView
import android.util.AttributeSet
import android.view.SurfaceHolder
import javax.microedition.khronos.egl.EGLConfig
import javax.microedition.khronos.opengles.GL10

/**
Created by Mr.Chan
Time 2024-07-29
Blog https://www.cnblogs.com/Frank-dev-blog/
 */
class PlayerSurface: GLSurfaceView, GLSurfaceView.Renderer, SurfaceHolder.Callback {
    private var onSurfaceCreated: (holder: SurfaceHolder) -> Unit = {}
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs){
        setRenderer(this)
    }

    public fun setOnSurfaceCreated(onSurfaceCreated: (holder: SurfaceHolder) -> Unit) {
        this.onSurfaceCreated = onSurfaceCreated
    }

    override fun surfaceCreated(holder: SurfaceHolder) {
        onSurfaceCreated(holder)
    }

    override fun surfaceChanged(holder: SurfaceHolder, format: Int, width: Int, height: Int) {
    }

    override fun surfaceDestroyed(holder: SurfaceHolder) {
    }

    override fun onDrawFrame(gl: GL10?) {
    }

    override fun onSurfaceChanged(gl: GL10?, width: Int, height: Int) {
    }

    override fun onSurfaceCreated(gl: GL10?, config: EGLConfig?) {

    }
}